tests/Fixer/StringNotation/StringImplicitBackslashesFixerTest.php

```diff
@@ -0,0 +1,878 @@
+            <<<'EOD'
+                <?php
+                $var = B"\A\a \' \8\9 \xZ \u";
+                $var = B"$foo \A\a \' \8\9 \xZ \u ${bar}";
+                EOD,
+            null,
+            ['double_quoted' => 'unescape'],
+        ];
+
+        yield [
+            <<<'EOD'
+                <?php
+                $var = B<<<HEREDOC
+                \A\Z
+                \a\z
+                \'
+                \8\9
+                \xZ
+                \u
+                HEREDOC;
+                $var = B<<<HEREDOC
+                $foo
+                \A\Z
+                \a\z
+                \'
+                \8\9
+                \xZ
+                \u
+                ${bar}
+                HEREDOC;
+
+                EOD,
+            null,
+            ['heredoc' => 'unescape'],
+        ];
+
+        yield [
+            <<<'EOD'
+                <?php
+                $var = "\\bar";
+                $var = "\\bar";
+                $var = "\\\\bar";
+                $var = "\\\\bar";
+                $var = "\\\\\\bar";
+                $var = "\\\\\\bar";
+                EOD,
+            <<<'EOD'
+                <?php
+                $var = "\bar";
+                $var = "\\bar";
+                $var = "\\\bar";
+                $var = "\\\\bar";
+                $var = "\\\\\bar";
+                $var = "\\\\\\bar";
+                EOD,
+        ];
+
+        yield [
+            <<<'EOD'
+                <?php
+                $var = '\\bar';
+                $var = '\\bar';
+                $var = '\\\\bar';
+                $var = '\\\\bar';
+                $var = '\\\\\\bar';
+                $var = '\\\\\\bar';
+                EOD,
+            <<<'EOD'
+                <?php
+                $var = '\bar';
+                $var = '\\bar';
+                $var = '\\\bar';
+                $var = '\\\\bar';
+                $var = '\\\\\bar';
+                $var = '\\\\\\bar';
+                EOD,
+            ['single_quoted' => 'escape'],
+        ];
+
+        yield [
+            <<<'EOD'
+                <?php
+                $var = <<<TXT
+                \\bar
+                \\bar
+                \\\\bar
+                \\\\bar
+                \\\\\\bar
+                \\\\\\bar
+                TXT;
+
+                EOD,
+            <<<'EOD'
+                <?php
+                $var = <<<TXT
+                \bar
+                \\bar
+                \\\bar
+                \\\\bar
+                \\\\\bar
+                \\\\\\bar
+                TXT;
+
+                EOD,
+        ];
+
+        yield [
+            <<<'EOD'
+                <?php
+                $var = <<<'TXT'
+                \bar
+                \\bar
+                \\\bar
+                \\\\bar
+                \\\\\bar
+                \\\\\\bar
+                TXT;
+
+                EOD,
+        ];
+
+        yield 'unescaped backslashes in single quoted string' => [
```

Comment:
Can we _not_ mix code snippets that should be ignored with ones that should be fixed? It's hard to grasp, I had to copy input/expected into Kdiff3 to see what's the difference between them:

<img width="721" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/b6cc19ce-3ad6-40e5-894d-53543f6ec400">

PS. Info for reviewers: cases above are more or less imported from `EscapeImplicitBackslashesFixerTest` (technical changes only, like heredoc closing markers), this case and cases below are new.

Reply:
Here we test 1..~8 backslashes, the test should stay as is, as in different styles (single quoted, double quoted, heredoc) it must behave differently with different escapes - but the input must be 1..~backslashes and +-same for all styles.

Reply:
I would argue that it would be better if each test case was split into 2 cases - one with lines that should remain untouched (and with `null` input, so Fixer's test case know that expected code is the same as input), second with lines that should be changed. Right now it's hard to see what is the purpose of the test case, because large part of input code remains untouched. Cognitive load would be much lower, because looking at the test case without input+expected I explicitly know nothing changes in the snippet. Looking at your test cases I don't know what is expected to change and what should be untouched. Each test case's snippet should be a minimal reproducer, without noise.

Reply:
I am the author who developed the fixer and these tests helped me as is. When the fixer behaves not as expected, IMHO these tests are much more easier to understand as is. If you want to have them in a different way, feel free to do so, but I do not support this change and once the PR is merged, these tests are not to be edit soon - user can always diff them.

Reply:
It does not matter who's the author in this case - fixer's code and its test cases must be understandable for all the maintainers, because at any point we can't demand you (or any other contributor) to provide fixes 🙂. For me these cases are not friendly and I don't want to need to copy and diff it externally to understand them. I believe that most of the tests merged during my tenure here had this convention because it's simply more readable (cases names starting with `ignore` + explicit untouched input).

Of course you can stick with your opinion, but then I can stick with my review state 😅. Especially that I don't have much time recently and I can't implement all my expectations on my own, because then I don't have time for my own ideas.

Reply:
```
                \
                \\
                \\\
                \\\\
                \\\\\
                \\\\\\
                \\\\\\\
```

Is a really consistent input noone can break, if we start splitting it, it will become really much harder to read, develop, and maintain, so please, reconsider this as answered. Thank you.

Reply:
> Is a really consistent input noone can break, if we start splitting it, it will become really much harder to read, develop, and maintain, so please, reconsider this as answered. Thank you.

I have totally different opinion about it, for me it's unreadable in current form because it's nearly impossible to grasp expected change by looking at the input/expected strings. @keradus @julienfalque @kubawerlos please share your thought about it is [this](https://github.com/mvorisek/PHP-CS-Fixer/blob/norm_escape_7621/tests/Fixer/StringNotation/StringImplicitBackslashesFixerTest.php#L664-L721) readable for you?

Reply:
done

Reply:
I still see some mixed test cases, where part of the snippet is fixed and part not, but at least these are not mixed that much. I can live with it, however in the future _please_ separate these 🙂. It's much more readable when the snippet contains only bare minimum required to show the case and focus is on _one_ behavior (either on _fix_ or on _ignoring_). Yields' keys also are very helpful to understand the purpose of the test case.